File: backend/app/prompts/emotion.py
"""情绪分析 Prompt。"""

EMOTION_ANALYSIS_SYSTEM_PROMPT = """你是心理健康情绪分析助手。根据用户对话或日记内容，输出结构化 JSON 情绪分析结果。

## 输出格式（严格 JSON，不要 markdown 代码块）
{
  "primaryEmotion": "主要情绪（如：焦虑、悲伤、平静、危机）",
  "emotionScore": 0.0到1.0的小数,
  "isNegative": true或false,
  "riskLevel": 0到3的整数,
  "keywords": ["关键词1", "关键词2"],
  "suggestion": "一句温暖的建议",
  "icon": "一个emoji",
  "label": "情绪标签描述",
  "riskDescription": "风险描述",
  "improvementSuggestions": ["改善建议1", "改善建议2", "改善建议3"]
}

## riskLevel 规则
- 0: 情绪稳定、积极
- 1: 轻度负面情绪
- 2: 明显焦虑/抑郁/压力，需关注
- 3: 危机信号（自伤/自杀/伤害他人），需立即干预

## 注意
- 只输出 JSON，不要有其他文字
- emotionScore 越高表示情绪强度越大
- 危机关键词：自杀、自残、不想活、结束生命、伤害他人
"""

EMOTION_ANALYSIS_USER_TEMPLATE = """请分析以下内容的情绪状态：

{content}

用户自评情绪：{dominant_emotion}
情绪评分（1-10）：{mood_score}
"""

SESSION_EMOTION_USER_TEMPLATE = """请分析以下心理咨询对话中用户情绪状态：

{content}
"""

File: src/utils/emotion.ts
import type { aiDataType } from '../types/emotionType'

const DEFAULT_AI_DATA: aiDataType = {
    primaryEmotion: '无分析数据',
    emotionScore: 0,
    isNegative: false,
    riskLevel: 0,
    keywords: [],
    suggestion: '',
    icon: '',
    label: '',
    riskDescription: '',
    improvementSuggestions: [],
    timestamp: 0,
}

/** 安全解析日记 AI 分析 JSON 字符串 */
export function parseAiEmotionAnalysis(raw: string | null | undefined): aiDataType {
    if (!raw || raw.trim() === '') return DEFAULT_AI_DATA
    try {
        const parsed = JSON.parse(raw)
        if (typeof parsed === 'object' && parsed !== null) {
            return { ...DEFAULT_AI_DATA, ...parsed }
        }
    } catch {
        // 非 JSON 格式，忽略
    }
    return DEFAULT_AI_DATA
}
